modules/users/server/controllers/users/users.profile.server.controller.js

```diff
@@ -95,7 +118,9 @@ exports.changeProfilePicture = function (req, res) {
 
   function updateUser() {
     return new Promise(function (resolve, reject) {
-      user.profileImageURL = config.uploads.profile.image.dest + req.file.filename;
+      user.profileImageURL = config.uploads.profile.s3Bucket ?
+        req.file.location :
+        '/' + req.file.path;
```

Comment:
This change is preventing the `deleteOldFile` from finding the existing file every time. I understand why this change was made. Otherwise, the front-end code wouldn't be able to find the location of the image. We might need to find a better way to handle this.

I'm on Windows, so it could be only happening in my environment. @Ghalleb Can you confirm if this is happening in your environment as well (if you're not on Windows, of course :) )

After adding a profile image, my user.profileImageUrl is `/modules\users\client\img\profile\uploads\12db59fbdd2193b3b8f99867e1640175`.

Reply:
To fix this issue, you can update this line https://github.com/meanjs/mean/blob/master/modules/users/server/controllers/users/users.profile.server.controller.js#L112 to resolve the path for the `unlink` method.

`fs.unlink(path.resolve('.' + existingImageUrl), function (unlinkError) {`

But it seems a little hacky to me.

Reply:
I'm not sure I understand what to do for this one.
On my environment (Ubuntu), everything is working fine.

For all the other changes, I will make them soon.

Thanks for the review.

Reply:
Can you verify that the existing file is getting deleted, and you're not getting an error logged to the console?

Reply:
I will...

Reply:
You are right, I have an error in the server logs:

`Removing profile image failed because file did not exist.`
`info: POST /api/users/picture 200 24.550 ms - 414`